ecl/hqlcpp/hqlhtcpp.cpp

```diff
@@ -5644,22 +5650,57 @@ void HqlCppTranslator::buildHashClass(BuildCtx & ctx, const char * name, IHqlExp
     endNestedClass(classStmt);
 }
 
-
-void HqlCppTranslator::buildCompareClass(BuildCtx & ctx, const char * name, IHqlExpression * sortList, const DatasetReference & dataset)
+void HqlCppTranslator::buildCompareClass(BuildCtx & ctx, const char * name, IHqlExpression * sortList, const DatasetReference & dataset, StringBuffer & compareFuncName)
 {
-    BuildCtx comparectx(ctx);
-    IHqlStmt * classStmt = beginNestedClass(comparectx, name, "ICompare");
-
+    if (options.useGlobalCompareClass)
     {
-        MemberFunction func(*this, comparectx, "virtual int docompare(const void * _left, const void * _right) const" OPTIMIZE_FUNCTION_ATTRIBUTE);
+        BuildCtx buildctx(*code, declareAtom);
+        GlobalCompareClass gcc(*this, sortList, dataset);
+
+        // stop duplicate classes being generated.
+        OwnedHqlExpr search = gcc.getUniqueKey();
+        HqlExprAssociation * match = buildctx.queryMatchExpr(search);
+        if (match)
+        {
+            match->queryExpr()->toString(compareFuncName);
+            return;
+        }
+
+        gcc.createGlobalCompareInstance(buildctx, name);
```

Comment:
Note compare functions are not being commoned up because of this - see aggsqx3, sortrev for example